ReactAndroid/build.gradle

```diff
@@ -318,6 +318,23 @@ android {
                     // This flag will suppress "fcntl(): Bad file descriptor" warnings on local builds.
                     arguments "--output-sync=none"
                 }
+
+                // Note: On Windows there are limits on number of character in file paths and in command lines
+                // Ref: https://android.googlesource.com/platform/ndk/+/master/docs/BuildSystemMaintainers.md#Path-Length-Limits
+                // NDK allows circumventing command line limits using response(RSP) files as inputs using NDK_APP_SHORT_COMMANDS flag.
+                // 
+                // Windows can support long file paths if configured through registry or by prefixing all file paths with a special character sequence 
+                // The latter requires changes in NDK. And there are tools in NDK (AR) which is not able to handle long paths (>256) even after setting the registry key.
+                // The new architecutre source tree is too deep, and the object file naming conventions in NDK makes the matters worse, by producing incredibly long file paths.
+                // Other solutions such as symlinking source code etc. didn't work as expected, and makes the build scripts complicated and hard to manage.
+                // This change temporarily works around the issue by placing the temporary build outputs as short a path as possible within the project path.
+                if (Os.isFamily(Os.FAMILY_WINDOWS) && 
+                    project.hasProperty("newArchEnabled") && 
+                    project.newArchEnabled == "true") {
+                    var rootProjPath = rootProject.projectDir.getParent().toString().trim()
```

Comment:
I'm not sure what was the intention here, but you're essentially picking the root folder of a RN project. This is risky as you're placing intermediate build artifacts inside a `.cxx` hidden folder in the root of the RN project.

Potentially this should be just something like: `$buildDir` (I would not go to the root folder as this is Android specific, unless you have a strong reason for that), so you'll be placing everything inside `$buildDir\\.cxx`.

Reply:
Essentially, i'm trying to place the object files in the shortest possible path within the root project .. Using $buildDir typically adds ~15 characters to the object file paths which makes them unavailable for llvm-aar ..

Reply:
I will need to debug llvm-ar to understand why it can't support long path .. May be it's fixed in a newer ndk ..

Reply:
> adds ~15 characters to the object file paths which makes them unavailable for llvm-aar ..

Could you show me an example of the resulting path? I'm trying to understand where are we hitting the limit.

I'd say that definitely this has to live inside the `android` folder (not in the top level). Maybe the `android/.cxx` folder could work. Still users on Win will suffer stale builds problems if this doesn't live inside the `build/` folder.

Reply:
This is an example path ..
D:\TestnewArc\node_modules\react-native\ReactAndroid\build\intermediates\cxx\Debug\351u4r48/obj/local/x86/objs-debug/rrc_switch/E_/gh-droid/testapp/test68/node_modules/react-native/ReactAndroid/__/ReactCommon/react/renderer/components/switch/androidswitch/react/renderer/components/androidswitch/AndroidSwitchShadowNode.o

This segment is ~230 characters;
"local/x86/objs-debug/rrc_switch/E_/gh-droid/testapp/test68/node_modules/react-native/ReactAndroid/__/ReactCommon/react/renderer/components/switch/androidswitch/react/renderer/components/androidswitch/AndroidSwitchShadowNode.o" .. and we have ~30 characters for the prefix ..